compiler/packages/babel-plugin-react-compiler/src/HIR/ObjectShape.ts

```diff
@@ -444,7 +444,281 @@ addObject(BUILTIN_SHAPES, BuiltInArrayId, [
       returnValueKind: ValueKind.Primitive,
     }),
   ],
-  // TODO: rest of Array properties
+  [
+    'copyWithin',
+    addFunction(BUILTIN_SHAPES, [], {
+      positionalParams: [],
+      restParam: Effect.Read,
+      returnType: {kind: 'Object', shapeId: BuiltInArrayId},
+      calleeEffect: Effect.Store,
+      returnValueKind: ValueKind.Mutable,
+    }),
+  ],
+  [
+    'fill',
+    addFunction(BUILTIN_SHAPES, [], {
+      positionalParams: [],
+      restParam: Effect.Capture,
+      returnType: {kind: 'Object', shapeId: BuiltInArrayId},
+      calleeEffect: Effect.Store,
+      returnValueKind: ValueKind.Mutable,
```

Comment:
this is subtle, but the behavior of `fill(value, start, end)` is that `value` is stored into the array (Effect.Capture), while the rest of the args are just read. So a more precise declaration would be:

```
positionalParams: [Effect.Capture],
restParam: Effect.Read,
```